modules/catalog/src/test/java/org/apache/ignite/internal/catalog/serialization/CatalogEntrySerializationCompatibilityTest.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.serialization;
+
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.assertThrows;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+import org.apache.ignite.internal.catalog.storage.VersionedUpdate;
+import org.apache.ignite.internal.util.io.IgniteDataInput;
+import org.apache.ignite.internal.util.io.IgniteDataOutput;
+import org.apache.ignite.lang.MarshallerException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests to verify catalog storage entries serialization compatibility.
+ */
+public class CatalogEntrySerializationCompatibilityTest {
+    private final CatalogEntrySerializerProvider serializerProvider = (ignore) -> TestUpdateEntrySerializer.INSTANCE;
```

Comment:
You have a single serializer version here.
I think the test idea is to have at least 2 versions, "old" and "new" ones. Where "old" one supports version 1 only, and "new" one supports version 1 and 2.

Reply:
From my point of view it doesn't affect the test case but breaks it. But for clarity, I divided it into 3 different serializers.